src/Task/AbstractTask.php

```diff
@@ -107,6 +108,7 @@ protected function defineOptions(array $options = [], array $additionalOptions =
         $optionsResolver->setAllowedTypes('execution_end_date', ['string', 'null']);
         $optionsResolver->setAllowedTypes('execution_start_time', [DateTimeImmutable::class, 'null']);
         $optionsResolver->setAllowedTypes('execution_end_time', [DateTimeImmutable::class, 'null']);
+        $optionsResolver->setAllowedTypes('execution_mode', ['string', 'null']);
```

Comment:
If the default value is `first_in_first_out`, no need to authorize `null`

Reply:
I wonder if this option is really necessary, as it is not used directly in the chainedTask. WDYT ?

Reply:
It depends if we move the responsibility of sorting tasks in the `ChainedTask` or `ChainedTaskBuilder`: 

- In the case of `ChainedTask`, we need the attribute to sort the tasks when retrieving it.
- In the case of `ChainedTaskBuilder`, we don't need the attribute as tasks are sorted when injected in `ChainedTask`

Reply:
IMHO, sorting tasks when the configuration is processed and tasks injected into a `ChainedTask` is not a good idea as "the configuration is law" BUT sorting tasks when retrieving the `ChainedTask` before executing it can be a way to go, in the end, sorting tasks is internal and only done for performances improvements.

Reply:
In the other end, is there any benefit of having a new attribute in `ChainedTask` just for sorting? That's a really weird use case IMHO

Reply:
I understand, orchestrate the tasks in the chainTasked means that it will be necessary to inject the ochestrator in the ChainedTask what is likely to be a problem during the serialization process when the Transporter creates the task no ?

Maybe sort them in the runner? 
https://github.com/Guikingone/SchedulerBundle/blob/0efd88bdd39c04dda6f9f821465f70f1305ea15a/src/Runner/ChainedTaskRunner.php#L41

Reply:
I think it's even worst to sort the tasks just before executing the whole list, the more I think about it, the more I like the initial approach where tasks are sorted during the `ChainedTask` construction phase, the configuration still "the law" but we adapt the internal structure to an improved one, this allows us to remove the `execution_mode` attribut in `AbstractTask` (which hide the sorting phase) and let the builder do the work.

Reply:
Another idea would be to do a recursive sort in the orchestrator - you wouldn't need to add the `execution_mode` option on the chained task - it would be the mode of the transporter that would be used. This is an idea, I don't really see how to do it.

Is a chained task nesting possible ? like this
```yaml
scheduler_bundle:
  transport:
    dsn: 'doctrine://default?execution_mode=first_in_first_out'
  tasks:
    chained:
      type: 'chained'
      tasks:
        world:
          type: 'command'
          command: 'app:world'
        hello:
          type: 'command'
          command: 'app:hello'
        sub_chained:
            type: 'chained'
            tasks:
              world:
                type: 'command'
                command: 'app:foo'
              hello:
                type: 'command'
                command: 'app:bar'
```

Reply:
That's an interesting idea but the limitation is defined via `DSN` which is not instantiated here, so sadly, we cannot access the `execution_mode` outside of the transport 🙁 

Maybe we can just sort the tasks when creating each one and let the builder call the policy orchestrator, it may seems strange but at the end, adding a new attribute in `ChainedTask` is not a big deal (I mean, that's not irrelevant): 

```php
public function __construct(string $name, string $executionMode, TaskInterface ...$tasks)
    {
        $this->defineOptions([
            'tasks' => $tasks,
            'execution_mode' => $executionMode,
        ], [
            'tasks' => ['SchedulerBundle\Task\TaskInterface[]', 'null'],
            'execution_mode' => 'string',
        ]);

        parent::__construct($name);
    }
```

This force to add both getter and setter and update the `TaskNormalizer` but at the end, is that really a big issue ? I don't think so 🤔 

This approach allows us to remove the `execution_mode` attribute from `AbstractTask`.

Reply:
I'm having trouble seeing how to call the orchestrator with the getExecutionMode() of the ChainedTask class.
Knowing that the sub-tasks are created in the ChainedBuilder at the ChainedTask instantiation here : 
https://github.com/Guikingone/SchedulerBundle/blob/0efd88bdd39c04dda6f9f821465f70f1305ea15a/src/Task/Builder/ChainedBuilder.php#L41-L51

Reply:
J'adore les échanges c'est très constructif ... mais je pense que me faire comprendre dans un anglais approximatif et corrigé par google traduction n'aide pas beaucoup 🤣

Reply:
No need to call the orchestrator in `ChainedTask` as the tasks are sorted during the creation process 🙂 

We can just update the builder: 

```php
unset($options['tasks'], $options['execution_mode']);
```

This way, we can get rid of the extra option 🙂